addons/sale_project/models/project.py

```diff
@@ -84,6 +84,16 @@ def _compute_sale_order_count(self):
         for project in self:
             project.sale_order_count = len(sale_order_items_per_project_id.get(project.id, self.env['sale.order.line']).order_id)
 
+    def _compute_invoice_count(self):
+        for project in self:
+            query = self.env['account.move.line']._search([('move_id.move_type', 'in', ['out_invoice', 'out_refund'])])
+            query.add_where('analytic_distribution ? %s', [str(project.analytic_account_id.id)])
+            query.order = None
+            query_string, query_param = query.select('COUNT(DISTINCT move_id)')
+            self._cr.execute(query_string, query_param)
+            count = self._cr.fetchone()[0]
+            project.invoice_count = count
```

Comment:
It is not really performant, we will do one query per project. 😬 
In your use case, indeed, it is enough but if one day we use it in kanban view or another view with many records, we will do many queries just for that. 😬 

Why not fetching everything and then processing the result in the for loop?

You can check how they did for the compute in AA as example. 🤔 

https://github.com/odoo/odoo/blob/c94d57edc002e499e317d04b2864d22acb5f40fa/addons/account/models/account_analytic_account.py#L18-L40